[#144] refactor: Car 도메인 코드 리팩토링

hyeonjinan096 opened the pull request:
close #144


## 💬 리뷰 요구사항(선택)

> Car 도메인 코드 리팩토링

- store, reader 분리
- CarFactory 구현

hyeonjinan096 committed:
[#144] refactor: Car 도메인 코드 리팩토링

- store, reader 분리
- CarFactory 구현

ijjung365 commented on rest/src/main/java/com/wherecar/rest/car/application/CarServiceImpl.java:
```diff
@@ -29,125 +28,76 @@ public class CarServiceImpl implements CarService {
+    public CarResponse createCar(Long companyId, CarRegisterRequest carRegisterRequest) {
 
         Company company = companyRepository.findById(companyId)
                 .orElseThrow(() -> new RuntimeException("Company 정보가 존재하지 않습니다."));
```
CompanyReader를 불러와서 company를 가져와주세요

hyeonjinan096 replied:
Todo 처리하고 지금 dev pull이 안되서 머지 후에 companyReader와 병합 되면 추후 처리해도 될까요?

ijjung365 replied:
네 알겠습니다

ijjung365 commented on rest/src/main/java/com/wherecar/rest/car/application/CarServiceImpl.java:
```diff
@@ -29,125 +28,76 @@ public class CarServiceImpl implements CarService {
-        car.changeMdn(carRegisterRequest.getMdn());
-        car.changeOwnerType(carRegisterRequest.getOwnerType());
-        car.changeAcquisitionType(carRegisterRequest.getAcquisitionType());
+        Car car = carRepository.findById(id).orElseThrow(() -> new RuntimeException("차량을 찾을 수 없습니다."));
```
CarStore 사용 부탁드립니다 repository도 service에서 바로 의존하면 안되요

ijjung365 commented on rest/src/main/java/com/wherecar/rest/car/application/CarServiceImpl.java:
```diff
@@ -29,125 +28,76 @@ public class CarServiceImpl implements CarService {
-                        .companyName(car.getCompany() != null ? car.getCompany().getName() : null)
-                        .build())
+        return carPage.getContent().stream()
+                .map(carFactory::toCarResponse)
```
이거 팩토리에 위임하시죠

ijjung365 commented on rest/src/main/java/com/wherecar/rest/car/infrastructure/infra/CarStoreImpl.java:
```diff
@@ -0,0 +1,27 @@
+    }
+
+    @Override
+    public void deleteById(Long id) {
```
그냥 delete가 좋을꺼같아요

devinjae reviewed (commented):
수정사항 확인 부탁드립니다

rjk0103 commented on rest/src/main/java/com/wherecar/rest/car/infrastructure/infra/CarStore.java:
```diff
@@ -0,0 +1,11 @@
+public interface CarStore {
+
+    Car store(Car car);
+    void deleteById(Long id);
```
id -> carId로 변경하는것이 더 좋을거 같습니다!!

rjk0103 reviewed (approved):
매개변수 Id값을 넘길때 carId로 명확하게 변경해주는것이 더 좋을거 같습니다.
고생하셨습니다.

hyeonjinan096 committed:
[#144] refactor: 불필요한 로그 삭제

hyeonjinan096 committed:
[#144] refactor: Car 도메인 2차 리팩토링 반영

- service에 있는 repository 제거
- carFactory에 toCarResponseList 메소드 추가
- deleteId -> delete 메소드 명 변경

devinjae reviewed (approved):
수고하셨습니다
